module/VuFind/src/VuFind/Controller/AbstractBaseFactory.php

```diff
@@ -41,6 +41,50 @@
  */
 class AbstractBaseFactory implements FactoryInterface
 {
+    /**
+     * Apply permission settings to the controller.
+     *
+     * @param ContainerInterface $container  Service manager
+     * @param AbstractBase       $controller Controller to configure
+     *
+     * @return AbstractBase
+     */
+    protected function applyPermissions($container, $controller)
+    {
+        $config = $container->get(\VuFind\Config\PluginManager::class)
+            ->get('permissionBehavior');
+        $permissions = $config->global->controllerAccess ?? [];
+        $setPermission = false;
```

Comment:
I'd prefer $permissionSet, otherwise it can be interpreted as an imperative further on.

Reply:
Further on, maybe the definition of $accessPermission could be refined to, say:

null = not set
false = set, no restrictions
string = permission

Note that the above use of false and null would be just the opposite of what currently happens.

Then we could check `$controller->getAccessPermission() === null` on line 79 and eliminate this variable altogether, no?